DSCResources/MSFT_xFirewall/MSFT_xFirewall.psm1

```diff
@@ -408,181 +552,62 @@ function Test-RuleProperties
-    }
-
-    if ($Action -and ($FirewallRule.Action -ne $Action))
-    {
-        Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
-            $($LocalizedData.PropertyNoMatchMessage) -f 'Action',$FirewallRule.Action,$Action
-            ) -join '')
-        $desiredConfigurationMatch = $false
-    }
-
-    if ($Profile)
-    {
-        [String[]] $networkProfileinRule = $FirewallRule.Profile.ToString() -replace(' ', '') -split(',')
-
-        if ($networkProfileinRule.Count -eq $Profile.Count)
+        $ParameterSource = (Invoke-Expression -Command "`$($($p.source))")
+        $ParameterNew = (Invoke-Expression -Command "`$$($p.name)")
+        switch ($p.type)
         {
-            foreach($networkProfile in $Profile)
-            {
-                if (-not ($networkProfileinRule -contains $networkProfile))
+            'String' {
+                # Perform a plain string comparison.
+                if ($ParameterNew -and ($ParameterSource -ne $ParameterNew))
                 {
                     Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
                         $($LocalizedData.PropertyNoMatchMessage) `
-                            -f 'Profile',($networkProfileinRule -join ','),($Profile -join ',')
+                            -f $p.Name,$ParameterSource,$ParameterNew
                         ) -join '')
                     $desiredConfigurationMatch = $false
-                    break
                 }
             }
-        }
-        else
-        {
-            Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
-                $($LocalizedData.PropertyNoMatchMessage) `
-                    -f 'Profile',($networkProfileinRule -join ','),($Profile -join ',')
-                ) -join '')
-            $desiredConfigurationMatch = $false
-        }
-    }
-
-    if ($Direction -and ($FirewallRule.Direction -ne $Direction))
-    {
-        Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
-            $($LocalizedData.PropertyNoMatchMessage) `
-                -f 'Direction',$FirewallRule.Direction,$Direction
-            ) -join '')
-        $desiredConfigurationMatch = $false
-    }
-
-    if ($RemotePort)
-    {
-        [String[]]$remotePortInRule = $properties.PortFilters.RemotePort
-
-        if ($remotePortInRule.Count -eq $RemotePort.Count)
-        {
-            foreach($port in $RemotePort)
-            {
-                if (-not ($remotePortInRule -contains($port)))
-                {
-                    Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
-                        $($LocalizedData.PropertyNoMatchMessage) `
-                            -f 'RemotePort',($remotePortInRule -join ','),($RemotePort -join ',')
-                        ) -join '')
-                    $desiredConfigurationMatch = $false
+            'Array' {
+                # Array comparison uses Compare-Object
+                if ($ParameterSource -eq $null) {
```

Comment:
Doh! I've been trying to make sure I do this but must have missed these. All fixed.